src/include/duckdb/common/types/row/tuple_data_segment.hpp

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/types/row/tuple_data_segment.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/mutex.hpp"
+#include "duckdb/common/unordered_set.hpp"
+#include "duckdb/common/vector.hpp"
+#include "duckdb/storage/buffer_manager.hpp"
+
+namespace duckdb {
+
+class TupleDataAllocator;
+class TupleDataLayout;
+
+struct TupleDataChunkPart {
+public:
+	TupleDataChunkPart();
+
+	//! Disable copy constructors
+	TupleDataChunkPart(const TupleDataChunkPart &other) = delete;
+	TupleDataChunkPart &operator=(const TupleDataChunkPart &) = delete;
+
+	//! Enable move constructors
+	TupleDataChunkPart(TupleDataChunkPart &&other) noexcept;
+	TupleDataChunkPart &operator=(TupleDataChunkPart &&) noexcept;
+
+	static constexpr const uint32_t INVALID_INDEX = (uint32_t)-1;
+
+public:
+	//! Index/offset of the row block
+	uint32_t row_block_index;
+	uint32_t row_block_offset;
+	//! Pointer/index/offset of the heap block
+	uint32_t heap_block_index;
+	uint32_t heap_block_offset;
+	data_ptr_t base_heap_ptr;
+	//! Size of the last heap row (for all but the last we can infer the size from the pointer difference)
```

Comment:
The comment suggests this is the entire heap, not just one row?

Reply:
Thanks, good catch! This comment is outdated. This now stores the heap size for this chunk part.